src/pool/srv_pool.c

```diff
@@ -2204,6 +2204,10 @@ pool_prop_read(struct rdb_tx *tx, const struct pool_svc *svc, uint64_t bits,
 	}
 
 	return 0;
+
+out_prop:
+	daos_prop_free(prop);
```

Comment:
Should *prop_out  = NULL here or *prop_out = prop should be put before return 0, otherwise might double free?

Reply:
Well, it should not happen as rc will always be negative when returning from here, and thus callers should not try to use prop_out, I believe ... But anyway, I agree this is much more clean ;-) Will push a new commit, thanks @wangshilong !!